### DISCUSSION TITLE
How to set labels on images defined by Generators

### DISCUSSION THREAD

**Discussion by davidecavestro** [eclipse/jkube#2884] [Q&A]
Hi, I want to migrate from _images_ to _generators_ the config for generating the image of a spring-boot app - as per [xml config docs](https://eclipse.dev/jkube/docs/kubernetes-maven-plugin/#_xml_configuration) - given that debugging jkube I saw Spring Boot layered jars are detected by the generator named _spring-boot_.

So this way I can control everything **except the labels of the generated image**, 
 [i.e. this way the labels are not set](https://github.com/davidecavestro/jkube/commit/a6a6b79051170c873b7f6ccaa3c1deffca42dd6b) and looking at jkube code I can't find any support for a way to define labels within the image through generators (please note here we aren't talking about labels of k8s descriptors)

```pom.xml
...
<generator>
  <config>
    <spring-boot>
      <color>always</color>
      <from>registry.access.redhat.com/ubi8/openjdk-17-runtime</from> <!-- control base for generated image name -->
      <name>my/image:0.0.1</name>                                     <!-- control generated image name -->
      <targetDir>/path/to/my/app</targetDir>                          <!-- control app folder within generated image -->
      <labels>foo=abc,bar=123</labels>                                <!-- THIS DOESN?T WORK: how can I do this? -->
    </spring-boot>
  </config>
</generator>
...
```
ps: changes depicted above on the spring-boot quickstart of the kubernetes maven plugin can be tested with `mvn package k8s:build -Pkubernetes`

**Comment by rohanKanojia** [comment]
I checked the code but it doesn't look like we support this functionality at the moment. 

I can see that in BaseGenerator we are adding some opinionated labels by checking the project configuration:

https://github.com/eclipse/jkube/blob/2ebeabe2ae6ebbcd52e0fd9323607892bcb831d4/jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/BaseGenerator.java#L258-L289

Shall we create an issue for this? I think we can plugin it some code to read labels configured by user in this method to get desired result.

**Reply by davidecavestro** [reply]
It would be great. From the UX perspective, it may be similar to the usage of _jkube.generator.tags_.
(please tell me if I have to create the issue)

**Reply by rohanKanojia** [reply]
@davidecavestro : that would be great!

**Reply by davidecavestro** [reply]
just opened https://github.com/eclipse/jkube/issues/2885
